### PR TITLE
[ci] Add tips for maintaining CI

### PR DESCRIPTION
**Pull request by markspanbroek**
Lessons learned from speeding up the CI in
https://github.com/status-im/nim-codex/pull/303

### CODE REVIEW

File: .github/workflows/Readme.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+### Steps for measuring time
+
+Breaking up steps allows you to see the time spent in each part. For instance,
+instead of having one step where all tests are performed, you might consider
+having separate steps for e.g. unit tests and integration tests, so that you can
+see how much time is spent in each.
+
+### Fix slow tests ###
+
+Try to avoid slow unit tests. They not only slow down continuous integration,
```
Comment (Menduist) [inline .github/workflows/Readme.md:46]: Useful: https://github.com/status-im/nim-unittest2/pull/12

Comment (markspanbroek) [reply .github/workflows/Readme.md:46]: Nice!

File: .github/workflows/Readme.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Performing git operations and compilation are both slow on Windows. This can
+easily mean that a Windows job takes twice as long as a Linux job. Therefore it
+makes sense to use a Windows runner only for testing Windows compatibility, and
+nothing else. Testing compatibility with other versions of Nim, code coverage
+analysis, etc. are therefore better performed on a Linux runner.
+
+### Parallelization ###
+
+Breaking up a long build job into several jobs that you run in parallel can have
```
Comment (Menduist) [inline .github/workflows/Readme.md:23]: I found this: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits
Not sure if that's our actual limit (no idea what our plan is), but it seems that the limitation is 20 concurrent jobs for the organization (or 5 for MacOS)

Comment (markspanbroek) [reply .github/workflows/Readme.md:23]: Thanks! I added a link to the usage limits in the readme.

**Pull request merged**